app/src/format.py

```diff
@@ -27,64 +28,100 @@ def format_guru_cards(
-            document = chunk_with_score.chunk.document
-            existing_doc = next(
-                (d for d in documents_with_scores if d.document == document),
-                None,
+    chunks_with_scores: list[ChunkWithScore],
+) -> OrderedDict[Document, list[ChunkWithScore]]:
+    chunks_with_scores.sort(key=lambda c: c.score, reverse=True)
+
+    # Build a dictionary of documents with their associated chunks,
+    # Ordered by the highest score of each chunk associated with the document
+    documents: OrderedDict[Document, list[ChunkWithScore]] = OrderedDict()
+    for chunk_with_score in chunks_with_scores[:docs_shown_max_num]:
+        document = chunk_with_score.chunk.document
+        if chunk_with_score.score < docs_shown_min_score:
+            logger.info(
+                "Skipping chunk with score less than %f: %s",
+                docs_shown_min_score,
+                chunk_with_score.chunk.document.name,
             )
-            if existing_doc:
-                existing_doc.max_score = max(existing_doc.max_score, chunk_with_score.score)
-            else:
-                documents_with_scores.append(DocumentWithMaxScore(document, chunk_with_score.score))
+            continue
 
-    # Sort the list by score
-    documents_with_scores.sort(key=lambda d: d.max_score, reverse=True)
+        if document in documents:
+            documents[document].append(chunk_with_score)
+        else:
+            documents[document] = [chunk_with_score]
 
-    # Only return the top docs_shown_max_num documents
-    return [d.document for d in documents_with_scores[:docs_shown_max_num]]
+    return documents
 
 
 def format_bem_documents(
     docs_shown_max_num: int,
     docs_shown_min_score: float,
-    chunks_with_scores: Sequence[ChunkWithScore],
+    chunks_with_scores: list[ChunkWithScore],
 ) -> str:
     documents = _get_bem_documents_to_show(
         docs_shown_max_num, docs_shown_min_score, chunks_with_scores
     )
-    formatted_documents = "".join([f"<li>{document.name}</li>" for document in documents])
-    return f"<h3>Source(s)</h3><ul>{formatted_documents}</ul>"
+
+    return _format_to_accordion_group_html(documents)
+
+
+def _format_to_accordion_html(document: Document, score: float) -> str:
+    global _accordion_id
+    _accordion_id += 1
+    similarity_score = f"<p>Similarity Score: {str(score)}</p>"
+
+    return f"""
+    <div class="usa-accordion" id=accordion-{_accordion_id}>
+        <h4 class="usa-accordion__heading">
+            <button
+                type="button"
+                class="usa-accordion__button"
+                aria-expanded="false"
+                aria-controls="a-{_accordion_id}"
+                >
+                <a href='https://link'>{document.name}</a>
+            </button>
+        </h4>
+        <div id="a-{_accordion_id}" class="usa-accordion__content usa-prose" hidden>
+            {"<p>Summary: " + document.content.strip() if document.content else ""}</p>
```

Comment:
If `not document.content`, then we probably don't want to show `Summary: `, right?
Also, is `Summary` the right descriptor since we're not summarizing the text?
Easy solution:
```suggestion
            {"<p>" + document.content.strip() if document.content else ""}</p>
```